pytorch_lightning/root_module/memory.py

```diff
@@ -50,10 +50,20 @@ def get_variable_sizes(self):
         input_ = self.model.example_input_array
 
         if self.model.on_gpu:
-            input_ = input_.cuda(0)
+            # test if input is a list or a tuple
+            if type(input_) is list or type(input_) is tuple:
+                    input_ = [input_i.cuda(0) if torch.is_tensor(input_i) else input_i 
+                              for input_i in input_]
+            else:
+                input_ = input_.cuda(0)
```

Comment:
I understand you didn't make this change, but always setting it to 0 might not be desired. Maybe to think about what this should be instead and fix here or in a followup

Reply:
I don't think it is an issue here. It is done just before training just to get the size of the input and output tensors of each layer. Distributing just makes it more complex.

Reply:
There is one other edge case outside of distributing (I agree with your sentiment there). For example I often do multiple runs on one GPU, and need to balance them in order to make sure I don't run out of memory. I'm not sure about the memory footprint of this (I'm thinking it's not a big deal), but I think it would be good practice to make sure that we always are deliberate about which GPUs we're on.

Reply:
Okay, i get your point. Would it be better to use something like this?
~~~ 
device = next(model.parameters()).get_device() 
~~~
We are at least making sure that the inputs will be on the same GPU as the first parameter in the model.

Reply:
That's perfect